stablehlo/reference/Ops.cpp

```diff
@@ -1023,14 +1040,17 @@ Tensor evalGatherOp(const Tensor &operand, const Tensor &startIndices,
 
   for (auto resultIt = result.index_begin(); resultIt != result.index_end();
        ++resultIt) {
+    auto resultIndex = *resultIt;
+
     Index batchIndex;
-    for (auto d : batchDims) batchIndex.push_back((*resultIt)[d]);
+    for (auto d : batchDims) batchIndex.push_back(resultIndex[d]);
 
-    Index startIndicesIndex = batchIndex;
+    auto startIndicesIndex = batchIndex;
     if (indexVectorDim < startIndices.getRank())
       startIndicesIndex.insert(startIndicesIndex.begin() + indexVectorDim,
                                kColon);
-    Index startIndex = evalIndex(evalSliceOp(startIndices, startIndicesIndex));
+
+    auto startIndex = evalIndex(evalSliceOp(startIndices, startIndicesIndex));
```

Comment:
Nit: I would remove the newline from here (and the same for scatter) to group `startIndicesIndex` and `startIndex` together, since startIndicesIndex is not an independent thing in the spec.